src/main/java/uk/gov/cshr/civilservant/controller/OrganisationalUnitController.java

```diff
@@ -40,12 +51,36 @@ public OrganisationalUnitController(OrganisationalUnitService organisationalUnit
     @GetMapping("/flat")
     @Cacheable("organisationalUnitsFlat")
     public ResponseEntity<List<OrganisationalUnitDto>> listOrganisationalUnitsAsFlatStructure() {
-        LOGGER.info("Getting org flat");
+        log.info("Getting org flat");
         List<OrganisationalUnitDto> organisationalUnitsMap = organisationalUnitService.getListSortedByValue();
 
         return ResponseEntity.ok(organisationalUnitsMap);
     }
 
+    @GetMapping("/flat/{domain}/")
+    public ResponseEntity<List<OrganisationalUnitDto>> listOrganisationalUnitsAsFlatStructureFilteredByDomainAndCode(@PathVariable String domain) {
+
+        log.info("Getting org flat, filtered by domain");
+        try {
+            List<OrganisationalUnit> organisationalUnits = organisationalUnitService.getOrganisationsForDomain(domain);
+            if (!organisationalUnits.isEmpty()) {
+                List<OrganisationalUnitDto> dtos = organisationalUnits.stream().map(ou -> organisationalUnitDtoFactory.create(ou)).collect(Collectors.toList());
```

Comment:
If you're doing a stream/map on a list, you might not need the `isEmpty` check on line 66

Reply:
Updated